modelon/impact/client/options.py

```diff
@@ -51,5 +62,82 @@ def with_values(self, **modified):
             sol_opts = custom_function.get_solver_options().with_values(rtol=1e-7)
             sim_opts = custom_function.get_simulation_options().with_values(ncp=500)
         """
-        values = _set_options(self._values, **modified,)
-        return ExecutionOptions(values, self._name, self._custom_func_sal)
+        values = _set_options(self._values, **modified)
+        return self.data(values)
+
+
+class CompilerOptions(BaseExecutionOptions):
+    def data(self, values):
+        """Returns a new CompilerOptions class instance.
+
+        Parameters:
+
+            values --
+                A keyworded, variable-length argument list of options.
+        """
+        return CompilerOptions(values, self._custom_function_name)
+
+
+class RuntimeOptions(BaseExecutionOptions):
+    def data(self, values):
+        """Returns a new RuntimeOptions class instance.
+
+        Parameters:
+
+            values --
+                A keyworded, variable-length argument list of options.
+        """
+        return RuntimeOptions(values, self._custom_function_name)
+
+
+class SimulationOptions(BaseExecutionOptions):
+    def data(self, values):
+        """Returns a new SimulationOptions class instance.
+
+        Parameters:
+
+            values --
+                A keyworded, variable-length argument list of options.
+        """
+        return SimulationOptions(values, self._custom_function_name,)
+
+    def with_result_filter(self, pattern: Union[str, List[str]]):
+        """Sets the variable filter for results.
+
+        Parameters:
+
+            parameters --
+                A keyworded, variable-length argument list of options.
```

Comment:
This parameter description will probably not help a user.

Maybe we could use the PyFMI docs a bit for inspiration to make the docs for this method better:
https://github.com/modelon-community/PyFMI/blob/ce3c42ff90e803d778b027e6d9e98b7f9bab8235/src/pyfmi/fmi_algorithm_drivers.py#L155